src/main/java/com/amazon/opendistroforelasticsearch/sql/plugin/RestSqlAction.java

```diff
@@ -67,8 +67,11 @@ public String getName() {
 
     @Override
     protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) {
+        SqlRequest sqlRequest = SqlRequest.NULL;
         try {
-            final SqlRequest sqlRequest = SqlRequestFactory.getSqlRequest(request);
+            sqlRequest = SqlRequestFactory.getSqlRequest(request);
```

Comment:
why not `final` anymore?

Reply:
> why not `final` anymore?

To make it `final`, it can be set only once (in `try` block) which is a compiling error because it's being read in `catch` block too.

Reply:
ok, assuming the `SqlRequestFactory.getSqlRequest` can throw, I see the problem now.

Reply:
Thanks! Resolving this if it makes sense.